bdai_ros2_wrappers/bdai_ros2_wrappers/action_handle.py

```diff
@@ -128,29 +140,21 @@ def get_feedback_callback(self, feedback: Action.Feedback) -> None:
 
     def _get_result_callback(self, future: Future) -> None:
         """Callback for pulling the resulting feedback out of the future and passing it to a user provided callback"""
-        result = future.result()
-        self._result = result.result
-
-        if result.status == GoalStatus.STATUS_SUCCEEDED:
-            self._logger.info("Finished successfully")
-            if self._wait_for_result_callback is not None:
-                self._wait_for_result_callback(True)
-            if self._result_callback is not None:
-                self._result_callback(self._result)
-            return
-        elif result.status == GoalStatus.STATUS_ABORTED:
-            self._logger.info("Aborted")
-            if self._wait_for_result_callback is not None:
-                self._wait_for_result_callback(False)
-            self._failure()
-            return
-        elif result.status == GoalStatus.STATUS_CANCELED:
-            self._logger.info("Canceled")
-            if self._wait_for_result_callback is not None:
-                self._wait_for_result_callback(False)
-            if self._on_cancel_success_callback is not None:
-                self._on_cancel_success_callback()
-            return
+        self._exception = future.exception()
+        if self._exception is None:
+            self._result = future.result()
+            if self._result.status == GoalStatus.STATUS_SUCCEEDED:
+                self._logger.info("Finished successfully")
+                if self._result_callback is not None:
+                    self._result_callback(self._result.result)
```

Comment:
This confuses me and did in the earlier instantiation of this as well.  If it's called `result_callback` shouldn't it be run on the return of a result whether or not it's successful?  Should the name actually be `success_callback`?  @amessing-bdai

Reply:
Probably should to be more clear. While we are discussing the specifics, it might also make sense to have different callbacks for rejected and aborted. Information can be passed back in `future.result().result` when an action is aborted. This information could be useful to a user.

Reply:
FTR I tried to keep APIs unchanged as much as possible to avoid major surgery downstream. In some cases I had to make changes as the implementation was inherently race-y. If we are up for an overhaul, I have some ideas :upside_down_face:

Reply:
AFAIK, SpotBasic is the only thing that uses this, so I am good with an overhaul. Really, we just want an easy way to deal with the lifecycle of an Action from the client side, an easy way to set specific callbacks, and overall to reduce duplication of boilerplate code from working with actions.

Reply:
Right, but that new API is likely to percolate up to `SpotBasic`, changing SpotBasic API itself. That's why I held it back.

Reply:
Do you mean the internals or the actual API? If it is just the internals and there is no regression, then I am still good with proceeding. If you are changing the actual API then we will need to discuss it. My current understanding is the constructor would change, but the actual function declarations would remain the same (based on https://github.com/bdaiinstitute/bdai/pull/927)?

Reply:
> My current understanding is the constructor would change, but the actual function declarations would remain the same (based on https://github.com/bdaiinstitute/bdai/pull/927)?

That is, if we keep APIs they way they are now. By API overhaul I mean, for example, changing how we deal with ongoing actions to be less callback-y and embrace asynchronous/synchronous programming. I left some ideas on the internal ticket behind this PR. We can continue the discussion there. No need to go forth with big changes here and now.